File: app/src/main/java/com/example/fw/fwsstopwatch/stats/STW.java
package com.example.fw.fwsstopwatch.stats;

/**
 * Created by fw on 27.11.16.
 */

public class STW implements Comparable<STW> {

    public final long start;
    public final long time;

    public STW(long start, long time) {
        this.start = start;
        this.time = time;
    }

    @Override
    public String toString() {
        return "STW{" +
                "start=" + start +
                ", time=" + time +
                '}';
    }

    @Override
    public int compareTo(STW another) {
        int dTime = (int) (this.time - another.time);
        if (dTime == 0) {
            return (int) (this.start - another.start);
        }
        return dTime;
    }
}

File: app/src/main/kotlin/com/fw/android/stw/service/STW.kt
package com.fw.android.stw.service

/**
 * Created by fw on 07.12.16.
 */
data class STW(val start: Long, val time: Long) : Comparable<STW> {
    val fmt: String = formatDateAndTime(start, time)

    override fun compareTo(other: STW): Int {
        val dTime = time - other.time
        val dStart = start - other.start
        return if (dTime != 0L) dTime.toInt() else dStart.toInt()
    }

}